README.md

```diff
@@ -1,32 +1,254 @@
-2. Invoking a code generation pipeline through
-   [artman]( https://github.com/googleapis/artman/#usage).
+### Prerequisites for code generation
 
-## The Google API Code Synchronizer
+1. You need a proto file describing your API. Proto files of existing Google APIs are available at
+[googleapis](https://github.com/googleapis/googleapis).
+  - Example: https://github.com/googleapis/googleapis/blob/master/google/cloud/language/v1/language_service.proto
+2. You need a yaml file describing certain service values. This is unfortunately not
+well-documented.
+  - Example: https://github.com/googleapis/googleapis/blob/master/google/cloud/language/language_v1.yaml
+3. Java 8 JDK
 
-The process currently hard-wires `kdiff3` for merging and conflict resolution.
-`kdiff3` runs in auto mode and will only pop up a UI if conflict resolution is
-not possible (note: X display needed in this case). There are two ways to invoke
-code synchronizer:
+### Process overview for generating a client library
 
-1. Invoking directly through a gradle task (see the build.gradle:runSynchronizer
-   task).
-2. Invoking a code generation pipeline which contains synchronization step
-   through [artman](https://github.com/googleapis/artman/#usage).
+1. Generate a descriptor file from the proto (once per API)
+2. Run client config generation (once per API)
+3. Manually tweak the generated client config
+4. Run code generation (once per API + language)
+
+Each of these steps are described in more detail below.
+
+### Set up prerequisites for building from source and running directly
+
+First, you need to install protoc if you don't have it yet: see
+[protocol-compiler-installation](https://github.com/google/protobuf#protocol-compiler-installation).
+The pre-built binaries are the easiest, but there aren't really instructions for installation -
+you have to just download the correct binaries for your platform at put them on your path.
+If you are using Linux x86_64:
+
+1. Download the latest zip from https://github.com/google/protobuf/releases ending in "-linux-x86_64.zip"
+2. Unzip the file, e.g. `unzip protoc-3.5.1-linux-x86_64.zip -d protoc-3.5.1`
+3. Copy protoc to a location on $PATH, e.g. `cp protoc-3.5.1/bin/protoc /usr/local/bin`
+
+Clone the present repository and build API Client Generator:
+
+```
+git clone https://github.com/googleapis/toolkit.git
+cd toolkit
+./gradlew fatJar
+```
+
+The googleapis/toolkit directory will hereafter be referenced as `${TOOLKIT_DIR}`.
+
+Clone the googleapis/googleapis repository (it has some config that is needed):
+
+```
+git clone https://github.com/googleapis/googleapis
+```
+
+The googleapis/googleapis directory will hereafter be referenced as `${GOOGLEAPIS_DIR}`.
+
+### Generate a descriptor file from the proto
+
+You need to locate/decide on the following before you can generate the descriptor file:
+
+1. The include directory with the proto files bundled with protoc (from the protoc setup step).
```

Comment:
But if I still keep the variable-like strings, I still need to define them...

I like the current ordering because it can just be followed in a sequence. This isn't quite a man page, it is a tutorial.

Reply:
Maybe, but the string "Hereafter, this will be referenced" has 15 exact matches in the readme... I would still prefer the reversed order. Even in the way people read this kind of "tutorials" they just want some command to copy paste and run, see if it fails and if it fails fix it and run again. Nobody really wants to understand details, just run the thing (so we show the command first and provide explanation only for those who want explanation, if the placeholder is descriptive enough one would not even need to read the explanation).